tfhe/src/integer/server_key/radix_parallel/scalar_add.rs

```diff
@@ -269,12 +276,560 @@ impl ServerKey {
             self.full_propagate_parallelized(ct);
         };
 
-        if self.is_eligible_for_parallel_single_carry_propagation(ct.blocks().len()) {
-            self.unchecked_scalar_add_assign(ct, scalar);
-            self.propagate_single_carry_parallelized(ct.blocks_mut())
+        let scalar_blocks = BlockDecomposer::new(scalar, self.message_modulus().0.ilog2())
+            .iter_as::<u8>()
+            .chain(std::iter::repeat(if scalar < Scalar::ZERO {
+                (self.message_modulus().0 - 1) as u8
+            } else {
+                0
+            }))
+            .take(ct.blocks().len())
+            .collect();
+
+        const COMPUTE_OVERFLOW: bool = false;
+        const INPUT_CARRY: bool = false;
```

Comment:
I used const over let, just because they are compile time constants, and I prefer let/const over direcly using the value in the argument as its more readable

Reply:
Ok, I tend to use const only for globals since rust vars are const by default, but that's fine too. I think it compiles to the same code in release anyways.